internal/csi-addons/rbd/replication.go

```diff
@@ -40,6 +41,12 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+// imageResyncMap is used to store the volumeID and image creation time for resync.
+// The volumeID gets added to the map when the DemoteVolume RPC is called and
+// removed from the map when the PromoteVolume RPC is called or
+// DisableVolumeReplication RPC is called.
+var imageResyncMap = sync.Map{}
```

Comment:
How does this get filled when the driver starts? Is it a problem if the state is not known from the beginning?

Reply:
This gets filled when cephcsi gets the Replication RPC calls, it's not a problem we might end up issuing rbd Resync (in case the CSI pods get restarted)

Reply:
ok, good to mention that in the comments too.

Reply:
i removed this logic and using image meta for this operation so that we will not have problem with restart